utils/src/main/java/org/jboss/windup/util/ExecutionStatistics.java

```diff
@@ -42,11 +44,49 @@ private ExecutionStatistics()
      */
     public static synchronized ExecutionStatistics get()
     {
-        if (stats.get() == null)
+        if (stats.get(Thread.currentThread()) == null)
+        {
+            stats.put(Thread.currentThread(),new ExecutionStatistics());
+        }
+        return stats.get(Thread.currentThread());
+    }
+
+    public static synchronized ExecutionStatistics get(Thread thread)
+    {
+        return stats.get(thread);
+    }
+
+
+    public void clear(Iterable<Thread> threads) {
+        for (Thread thread : threads)
+        {
+            stats.remove(thread);
+        }
+    }
+
+    public ExecutionStatistics mergeSiblingThreads() {
+        int numberOfSiblingThreads = Thread.currentThread().getThreadGroup().activeCount();
+        Thread[] threads = new Thread[numberOfSiblingThreads];
+        Thread.currentThread().getThreadGroup().enumerate(threads);
+        for(int i=0;i<threads.length;i++) {
+            merge(stats.get(threads[i]));
```

Comment:
this should iterate over all the threads that were created underneath the main windup thread.. However I need to check if dead threads are preserved in this enumeration.